### PR TITLE
added a new event 'progress'

### PR DESCRIPTION
**Pull request by honnix**
@ulzha PTAL

## Description

Added a new event named `event.core.progress`.

## Motivation and Context

This event can be fired by the task itself while running. The purpose is
for the task to report progress, metadata or any generic info so that
event handler listening for this can keep track of the progress of
running task.

## Have you tested this? If so, how?
 

I have included unit tests.

### CODE REVIEW

**Comment by ulzha** [comment]
LGTM

**Comment by ulzha** [comment]
One urging side thought - there is some existing code already, catering for similar communication:
* [`TaskStatusReporter`](https://github.com/spotify/luigi/blob/de4629a4c7c562b53fee2045aabd84189f897593/luigi/worker.py#L288), that has ` update_progress_percentage` method, among others, for sending some communication from the task
* [`SchedulerMessage`](https://github.com/spotify/luigi/blob/de4629a4c7c562b53fee2045aabd84189f897593/luigi/worker.py#L319), that allows receiving communication from the scheduler by the task

I guess anyway the use of events would be preferable, could allow us to eventually replace the ad-hoc code?

**Comment by honnix** [comment]
Are those for communication from worker to scheduler? If that is the case, it doesn't seem to fit very well to our use case.

**Comment by honnix** [comment]
@Tarrasch 👍 for that, will fix.

File: luigi/event.py [hunk 1]
```diff
@@ -25,6 +25,10 @@ class Event(object):
     DEPENDENCY_PRESENT = "event.core.dependency.present"
     BROKEN_TASK = "event.core.task.broken"
     START = "event.core.start"
+    #: This event can be fired by the task itself while running. The purpose is
```
Comment (honnix) [inline luigi/event.py]: Tried following existing inline doc. How does it look? @Tarrasch

**Comment by ulzha** [comment]
I didn't mean to include communication with the scheduler in our solution. It was just an aside comment, that we'll now be "reporting progress" in 2 different ways, and that I prefer event approach among those.

**Comment by honnix** [comment]
OK. Got it.

**Pull request merged**